pkg/ottl/parser.go

```diff
@@ -252,6 +262,10 @@ func NewStatementSequence[K any](statements []*Statement[K], telemetrySettings c
 		statements:        statements,
 		errorMode:         PropagateError,
 		telemetrySettings: telemetrySettings,
+		tracer:            &noop.Tracer{},
```

Comment:
setting the tracer to `noop.Tracer` here to avoid nil pointer access in case no tracer is set telemetrySettings - this seemingly happens at the startup of the collector when the self monitoring feature flag is enabled